lib/internal/Magento/Framework/App/DeploymentConfig/Writer.php

```diff
@@ -131,7 +131,7 @@ public function saveConfig(array $data, $override = false, $pool = null, array $
                 if ($currentData) {
                     if ($override) {
                         $config = array_merge($currentData, $config);
-                    } else {
+                    } elseif (is_array($currentData)) {
```

Comment:
How `$currentData` equal to `null` passes condition in line 131?

Reply:
Changed the description wording to use "not an array".

Reply:
Now the static analysis complains. I guess I'll move the condition up to 131.